### PR TITLE
Add names and avatars

### PR DESCRIPTION
**Pull request by marcindbar**
[Ticket](https://kindergarten.stxnext.pl/browse/WDP-88)
[Pylint & tests](https://gist.github.com/marcindbar/0d413e92d8dd5b5310aaa83419a1e827)

### CODE REVIEW

File: src/presence_analyzer/script.py [hunk 1]
```diff
@@ -111,3 +112,15 @@ def action_stop(dry_run=False):
         _serve('stop', dry_run=dry_run)
 
     werkzeug.script.run()
+
+
+# bin/overwrite-users-data
+def overwrite_users_data():
```
Comment (dpolcyn) [inline src/presence_analyzer/script.py]: update_user_data

File: src/presence_analyzer/static/js/mean.js [hunk 2]
```diff
@@ -6,8 +6,10 @@
 
         $('#user-id').change(function(){
             var $chartDiv = $('#chart-div'),
+                errorMessage = document.getElementById('data-error');
```
Comment (dpolcyn) [inline src/presence_analyzer/static/js/mean.js]: , instead of ;

File: src/presence_analyzer/tests.py [hunk 3]
```diff
@@ -54,7 +60,20 @@ def test_api_users(self):
         self.assertEqual(resp.status_code, 200)
         self.assertEqual(resp.content_type, 'application/json')
         self.assertEqual(len(data), 2)
-        self.assertDictEqual(data[0], {u'user_id': 10, u'name': u'User 10'})
+        self.assertDictEqual(
+            data[0],
+            {
+                u'user_id': 10,
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: from __future__ import unicode_iterals then you don't have use u'

File: src/presence_analyzer/static/js/helpers.js [hunk 4]
```diff
@@ -24,3 +24,27 @@ function drawChart(chartDiv, loading, chart, data, options) {
     loading.hide();
     chart.draw(data, options);
 }
+
+function showError(jqXHR, loading, message) {
+    loading.hide();
+    if (jqXHR.status == 404) {
```
Comment (piotrgrundas) [inline src/presence_analyzer/static/js/helpers.js]: always use` ===`

File: src/presence_analyzer/static/js/helpers.js [hunk 5]
```diff
@@ -24,3 +24,27 @@ function drawChart(chartDiv, loading, chart, data, options) {
...
+        message.textContent = 'There is no data available.';
+    }
+}
+
+(function($) {
+    $(document).ready(function() {
+        $('#user-id').change(function() {
+            var avatarUrl = $('#avatar'),
+                selectedUser = $('#user-id').val(),
+                userName = $('#user-name');
```
Comment (piotrgrundas) [inline src/presence_analyzer/static/js/helpers.js]: $userName, $avatarUrl

File: src/presence_analyzer/static/js/helpers.js [hunk 6]
```diff
@@ -24,3 +24,27 @@ function drawChart(chartDiv, loading, chart, data, options) {
...
+
+            avatarUrl.hide();
+            userName.hide();
+
+            $.getJSON('/api/v1/users/' + selectedUser, function(result) {
+                avatarUrl.empty().prepend('<img src="' + result.avatar + '" />').show();
```
Comment (piotrgrundas) [inline src/presence_analyzer/static/js/helpers.js]: `$('', {src: xxx});`

File: src/presence_analyzer/static/js/mean.js [hunk 7]
```diff
@@ -6,8 +6,10 @@
 
         $('#user-id').change(function(){
             var $chartDiv = $('#chart-div'),
+                errorMessage = document.getElementById('data-error'),
```
Comment (piotrgrundas) [inline src/presence_analyzer/static/js/mean.js]: Why since ur using every where else $?

File: src/presence_analyzer/utils.py [hunk 8]
```diff
@@ -126,6 +127,91 @@ def group_by_weekday_start_end(items):
...
+    return {
+        'id': int(user.attrib['id']),
+        'data': {
+            'name': user.find('name').text,
+            'avatar': '{}{}'.format(url, user.find('avatar').text)
+        }
+    }
+
+
+def create_avatar_address(server):
```
Comment (piotrgrundas) [inline src/presence_analyzer/utils.py]: 1) server_addres IMO
2) Is this func necessary? Ur using it only once so I would skip this one and create this url directly in place where r u using it.

File: src/presence_analyzer/utils.py [hunk 9]
```diff
@@ -126,6 +127,91 @@ def group_by_weekday_start_end(items):
...
+        'user_id_0': {
+            'name': 'Kajetan O.',
+            'avatar': 'https://intranet.stxnext.pl:443/api/images/users/130',
+        },
+        'user_id_1': {
+            'name': 'Adam P.',
+            'avatar': 'https://intranet.stxnext.pl:443/api/images/users/141',
+        },
+        ...
+    }
```
Comment (piotrgrundas) [inline src/presence_analyzer/utils.py]:
```
Creates a dictionary with users full info.

Rreturns: (dict) - with user data, like:
    {
        'user_id_0': {
            'name': 'Kajetan O.',
            'avatar': 'https://intranet.stxnext.pl:443/api/images/users/130',
        },
        'user_id_1': {
            'name': 'Adam P.',
            'avatar': 'https://intranet.stxnext.pl:443/api/images/users/141',
        },
        ...
    }
```

**Pull request merged**